src/tests/libecalc/dto/test_categories.py

```diff
@@ -3,96 +3,91 @@
         with pytest.raises(ValidationError) as exc_info:
-            dto.components.VentingEmitter(
+            YamlVentingEmitter(
                 name="test",
-                regularity={datetime(2000, 1, 1): Expression.setup_from_expression(1)},
-                emission_name="CH4",
-                emitter_model={datetime(2000, 1, 1): emitter_model},
-                user_defined_category={datetime(2000, 1, 1): "VENTING-EMISSIONS"},
+                emission=emission,
+                user_defined_category="VENTING-EMISSIONS",
             )
         exception: ValidationError = typing.cast(ValidationError, exc_info.value)
         assert (
             exception.errors()[0]["msg"]
-            == "CATEGORY: VENTING-EMISSIONS is not allowed for VentingEmitter with the name test. Valid categories are: ['BASE-LOAD', 'COLD-VENTING-FUGITIVE', 'COMPRESSOR', 'FIXED-PRODUCTION-LOAD', 'FLARE', 'MISCELLANEOUS', 'PUMP', 'GAS-DRIVEN-COMPRESSOR', 'TURBINE-GENERATOR', 'POWER-FROM-SHORE', 'OFFSHORE-WIND', 'LOADING', 'STORAGE', 'STEAM-TURBINE-GENERATOR', 'BOILER', 'HEATER']"
+            == "CATEGORY: VENTING-EMISSIONS is not allowed for YamlVentingEmitter with the name test. Valid categories are: ['BASE-LOAD', 'COLD-VENTING-FUGITIVE', 'COMPRESSOR', 'FIXED-PRODUCTION-LOAD', 'FLARE', 'MISCELLANEOUS', 'PUMP', 'GAS-DRIVEN-COMPRESSOR', 'TURBINE-GENERATOR', 'POWER-FROM-SHORE', 'OFFSHORE-WIND', 'LOADING', 'STORAGE', 'STEAM-TURBINE-GENERATOR', 'BOILER', 'HEATER']"
         )
 
         # Check that lower case raises error
         with pytest.raises(ValidationError) as exc_info:
-            dto.components.VentingEmitter(
+            YamlVentingEmitter(
                 name="test",
-                regularity={datetime(2000, 1, 1): Expression.setup_from_expression(1)},
-                emission_name="CH4",
-                emitter_model={datetime(2000, 1, 1): emitter_model},
-                user_defined_category={datetime(2000, 1, 1): "fuel-gas"},
+                emission=emission,
+                user_defined_category="fuel-gas",
             )
 
         exception: ValidationError = typing.cast(ValidationError, exc_info.value)
         assert (
             exception.errors()[0]["msg"]
-            == "CATEGORY: fuel-gas is not allowed for VentingEmitter with the name test. Valid categories are: ['BASE-LOAD', 'COLD-VENTING-FUGITIVE', 'COMPRESSOR', 'FIXED-PRODUCTION-LOAD', 'FLARE', 'MISCELLANEOUS', 'PUMP', 'GAS-DRIVEN-COMPRESSOR', 'TURBINE-GENERATOR', 'POWER-FROM-SHORE', 'OFFSHORE-WIND', 'LOADING', 'STORAGE', 'STEAM-TURBINE-GENERATOR', 'BOILER', 'HEATER']"
+            == "CATEGORY: fuel-gas is not allowed for YamlVentingEmitter with the name test. Valid categories are: ['BASE-LOAD', 'COLD-VENTING-FUGITIVE', 'COMPRESSOR', 'FIXED-PRODUCTION-LOAD', 'FLARE', 'MISCELLANEOUS', 'PUMP', 'GAS-DRIVEN-COMPRESSOR', 'TURBINE-GENERATOR', 'POWER-FROM-SHORE', 'OFFSHORE-WIND', 'LOADING', 'STORAGE', 'STEAM-TURBINE-GENERATOR', 'BOILER', 'HEATER']"
```

Comment:
Is there a way we could keep the VentingEmitter name here? Are we using `type`when creating the error? In pydantic we can specify 
```
class Config:
  title = "VentingEmitter"
```
to manipulate the title when generating json-schema, but not sure how we can use that name when generating errors.

Reply:
This is fixed now, the title is used when generating the error.